internal/graph/check.go

```diff
@@ -221,6 +239,7 @@ func union(ctx context.Context, concurrencyLimit uint32, handlers ...CheckHandle
 				err = result.err
 				continue
 			}
+			dbReads += result.resp.GetResolutionMetadata().DatastoreCallCount
 
 			if result.resp.GetAllowed() {
 				return result.resp, nil
```

Comment:
Shouldn't the response returned here reference the value of `dbReads`?

Let's say you have a union of 2 check handlers here, the first handler takes 2 database queries and doesn't find an answer and the second handler takes 2 database queries and finds an answer. The result should be 4 total database queries at the time we respond, but this will only return 2.

Seems this should be:
```go
if result.resp.GetAllowed() {
    return &ResolveCheckResponse{
        Allowed: true, 
        ResolutionMetadata: &ResolutionMetadata{
	    DatastoreCallCount: dbReads,
        },
    }
```

I may be overlooking something though? Could you clarify.

Reply:
Honestly I went back and forth with how to calculate the final number. I think at some point I experimented with this idea but I didn't get the tests to pass, most likely because I didn't figure out til later that TTU calculation is different than the rest. (I had the problem that I was over-counting).

I made the changes now, let me know if you agree with the results of the unit tests 👍

Reply:
Looks good to me now! Thanks for the updates.